### PR TITLE
[mod_articles_category] Option to group by tags

### PR DESCRIPTION
**Pull request by SharkyKZ**
Pull Request for Issue #18458 .

### Summary of Changes

This adds an option to group articles by tags.

### Testing Instructions

Create `Articles - Category` module. Set `Article Grouping` to `Tags`.

### Expected result

Articles grouped by tags. Articles without tags are placed in `Untagged` group.

### Documentation Changes Required
New option added.

### CODE REVIEW

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on d04ede48ad714bff6573d9b4505ace00bb3ddfb1<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21083">issues.joomla.org/tracker/joomla-cms/21083</a>.</sub>

File: modules/mod_articles_category/mod_articles_category.php [hunk 1]
```diff
@@ -88,6 +88,9 @@
 			case 'category_title' :
 				$list = ModArticlesCategoryHelper::groupBy($list, $article_grouping, $article_grouping_direction);
 				break;
+			case 'tags':
```
Comment (Quy) [inline modules/mod_articles_category/mod_articles_category.php]: Add a space before :

File: modules/mod_articles_category/mod_articles_category.php [hunk 2]
```diff
@@ -88,6 +88,9 @@
 			case 'category_title' :
 				$list = ModArticlesCategoryHelper::groupBy($list, $article_grouping, $article_grouping_direction);
 				break;
+			case 'tags' :
+				$list = ModArticlesCategoryHelper::groupByTags($list, $article_grouping_direction);
+				break;
 			default:
```
Comment (Quy) [inline modules/mod_articles_category/mod_articles_category.php:94]: Add space before :

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 1887d48fd5235459b8a30885deca34016b6ff3ce<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21083">issues.joomla.org/tracker/joomla-cms/21083</a>.</sub>

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on 1887d48fd5235459b8a30885deca34016b6ff3ce<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21083">issues.joomla.org/tracker/joomla-cms/21083</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
Ready to Commit after two successful tests.

**Comment by ghost** [comment]
Ready to Commit after two successful tests.

**Pull request merged**